src/common/blkdev.cc

```diff
@@ -181,82 +148,70 @@ int64_t get_block_device_int_property(const char *devname, const char *property)
   return r;
 }
 
-bool block_device_support_discard(const char *devname)
+bool BlkDev::block_device_support_discard()
 {
-  return get_block_device_int_property(devname, "queue/discard_granularity") > 0;
+  return get_block_device_int_property(BLKDEV_PROP_DISCARD_GRANULARITY) > 0;
 }
 
-int block_device_discard(int fd, int64_t offset, int64_t len)
+int BlkDev::block_device_discard(int64_t offset, int64_t len)
 {
   uint64_t range[2] = {(uint64_t)offset, (uint64_t)len};
   return ioctl(fd, BLKDISCARD, range);
 }
 
-bool block_device_is_rotational(const char *devname)
+bool BlkDev::block_device_is_nvme()
 {
-  return get_block_device_int_property(devname, "queue/rotational") > 0;
+  char vendor[80];
+  // nvme has a device/device/vendor property; infer from that.  There is
+  // probably a better way?
+  int r = get_block_device_string_property(BLKDEV_PROP_VENDOR, vendor, 80);
+  return (r > 0);
```

Comment:
`get_block_device_string_property` returns 0 on success.

Reply:
@asomers this is not addressed.

Reply:
@tchaikov it is addressed; you're probably only looking at a subset of the changes.